dotnet/samples/KernelSyntaxExamples/Example59_OpenAIFunctionCalling.cs

```diff
@@ -93,26 +93,35 @@ private static async Task CompleteChatWithFunctionsAsync(string ask, ChatHistory
             // you can invoke it using the following code.
             if (kernel.Functions.TryGetFunctionAndContext(functionResponse, out ISKFunction? func, out ContextVariables? context))
             {
-                var kernelResult = await kernel.RunAsync(func, context);
+                var result = (await kernel.RunAsync(func, context)).GetValue<object>();
```

Comment:
Do we intentionally skip adding an LLM function call request to the chat history before adding the function execution result?

<img width="386" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/9c44c966-8be6-495f-bc5f-dc2996c2ff36">
Link - https://platform.openai.com/docs/guides/gpt/function-calling

Reply:
Good catch, I think i just overlooked that. However, now that you've pointed it out, it's not as simple as it seems to add the OpenAI message back into chat history. We can access the message via `IChatResult.ModelResult.GetResult<ChatModelResult>().Choice.Message`, which gives us back an `Azure.AI.OpenAI.ChatMessage` object. However, I dont believe we have the capability to add that back into chat history without losing some information in the translation (would convert to our `ChatMessage` to add to `ChatHistory`, and then back to `Azure.AI.OpenAI.ChatMessage` in `ClientBase.cs`)

Reply:
Actually.... I think we are able to handle this with the dictionary approach. When adding this message back into chat history, would need to add key/value for "function_call" into the dictionary and make sure that gets copied back into the OAI message. I will update the PR with this.